src/darwin/Framework/CHIPTests/TestHelpers/MTRDeviceTestDelegate.h

```diff
@@ -23,31 +23,31 @@ typedef void (^MTRDeviceTestDelegateDataHandler)(NSArray<NSDictionary<NSString *
-@property (nonatomic, nullable) MTRDeviceTestDelegateDataHandler onAttributeDataReceived;
-@property (nonatomic, nullable) MTRDeviceTestDelegateDataHandler onEventDataReceived;
-@property (nonatomic, nullable) dispatch_block_t onReportBegin;
-@property (nonatomic, nullable) dispatch_block_t onReportEnd;
-@property (nonatomic, nullable) dispatch_block_t onDeviceCachePrimed;
-@property (nonatomic) BOOL skipExpectedValuesForWrite;
-@property (nonatomic) BOOL forceAttributeReportsIfMatchingCache;
-@property (nonatomic, nullable) dispatch_block_t onDeviceConfigurationChanged;
-@property (nonatomic) BOOL pretendThreadEnabled;
-@property (nonatomic, nullable) dispatch_block_t onSubscriptionPoolDequeue;
-@property (nonatomic, nullable) dispatch_block_t onSubscriptionPoolWorkComplete;
-@property (nonatomic, nullable) dispatch_block_t onClusterDataPersisted;
-@property (nonatomic, nullable) dispatch_block_t onSubscriptionCallbackDelete;
-@property (nonatomic, nullable) dispatch_block_t onSubscriptionReset;
-@property (nonatomic, nullable) NSNumber * subscriptionMaxIntervalOverride;
-@property (nonatomic, nullable) MTRDeviceTestDelegateHandler onUTCTimeSet;
-@property (nonatomic) BOOL forceTimeUpdateShortDelayToZero;
-@property (nonatomic) BOOL forceTimeSynchronizationLossDetectionCadenceToZero;
+@property (atomic, nullable) dispatch_block_t onReachable;
+@property (atomic, nullable) dispatch_block_t onNotReachable;
+@property (atomic, nullable) dispatch_block_t onInternalStateChanged;
+@property (atomic, nullable) MTRDeviceTestDelegateDataHandler onAttributeDataReceived;
+@property (atomic, nullable) MTRDeviceTestDelegateDataHandler onEventDataReceived;
+@property (atomic, nullable) dispatch_block_t onReportBegin;
+@property (atomic, nullable) dispatch_block_t onReportEnd;
+@property (atomic, nullable) dispatch_block_t onDeviceCachePrimed;
+@property (atomic) BOOL skipExpectedValuesForWrite;
+@property (atomic) BOOL forceAttributeReportsIfMatchingCache;
+@property (atomic, nullable) dispatch_block_t onDeviceConfigurationChanged;
+@property (atomic) BOOL pretendThreadEnabled;
+@property (atomic, nullable) dispatch_block_t onSubscriptionPoolDequeue;
+@property (atomic, nullable) dispatch_block_t onSubscriptionPoolWorkComplete;
+@property (atomic, nullable) dispatch_block_t onClusterDataPersisted;
+@property (atomic, nullable) dispatch_block_t onSubscriptionCallbackDelete;
+@property (atomic, nullable) dispatch_block_t onSubscriptionReset;
+@property (atomic, nullable) NSNumber * subscriptionMaxIntervalOverride;
+@property (atomic, nullable) MTRDeviceTestDelegateHandler onUTCTimeSet;
+@property (atomic) BOOL forceTimeUpdateShortDelayToZero;
+@property (atomic) BOOL forceTimeSynchronizationLossDetectionCadenceToZero;
 @end
 
 @interface MTRDeviceTestDelegateWithSubscriptionSetupOverride : MTRDeviceTestDelegate
-@property (nonatomic) BOOL skipSetupSubscription;
+@property (atomic) BOOL skipSetupSubscription;
```

Comment:
Adding a conversation to prevent auto merge. I will resolve this conversation when there's been a few CI runs without hitting the `test048_MTRDeviceResubscribeOnSubscriptionPool` race.